siriuspy/siriuspy/clientarch/time.py

```diff
@@ -1,84 +1,51 @@
 """Time conversion module."""
 
-from datetime import datetime as _datetime
+from datetime import datetime as _datetime, timedelta as _timedelta
 
 
-class Time:
+class Time(_datetime):
     """Time conversion class."""
 
-    def __init__(self, year=None, month=None, day=None,
-                 hour=None, minute=0, second=0, microsecond=0,
-                 timestamp=None, datetime=None,
-                 timestamp_string=None, timestamp_format=None):
-        """Init."""
-        if datetime:
-            if not isinstance(datetime, _datetime):
-                raise TypeError("Expected an argument of type 'datetime'.")
-            self.datetime = datetime
-        elif timestamp:
+    def __new__(cls, year=None, month=None, day=None,
+                hour=None, minute=0, second=0, microsecond=0,
+                timestamp=None,
+                timestamp_string=None, timestamp_format=None):
+        """New."""
+        if timestamp:
```

Comment:
I think `timestamp` is the name of the string, I think it is confusing to use it as a `float` here. Maybe the name of the `float` argument could be `time`?

Also, wouldn't it be useful if the `time` argument were the first keyword argument, such that it would be possible to call:
`Time(time.time())`

What happens if we call `Time()`? Is this equivalent to `Time.now()`?

Reply:
> What happens if we call `Time()`? Is this equivalent to `Time.now()`?

No, in the current implementation this sentence returns an error, as like the super class. Do you think it is useful/intuitive this implementation?

Reply:
(offline talk summary)
We decided to keep `timestamp` name for float inputs and `timestamp_string` for string inputs to maintain consistency with the superclass nomenclature

Reply:
> Also, wouldn't it be useful if the time argument were the first keyword argument, such that it would be possible to call:
Time(time.time())

done!